cmd/earth/main.go

```diff
@@ -298,8 +299,33 @@ func (app *earthApp) parseConfigFile(context *cli.Context) error {
 		cfg.Git = map[string]config.GitConfig{}
 	}
 
+	err = app.processDeprecatedCommandOptions(context, cfg)
+	if err != nil {
+		return err
+	}
+
+	gitConfig, gitCredentials, err := config.CreateGitConfig(cfg)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create git config from %s", app.configPath)
+	}
+
+	// command line option overrides the config which overrides the default value
+	if !context.IsSet("buildkit-image") && cfg.Global.BuildkitImage != "" {
+		app.buildkitdImage = cfg.Global.BuildkitImage
+	}
+
+	app.buildkitdSettings.TempDir = cfg.Global.CachePath
+	app.buildkitdSettings.GitConfig = gitConfig
+	app.buildkitdSettings.GitCredentials = gitCredentials
+	return nil
+
+}
+
+func (app *earthApp) processDeprecatedCommandOptions(context *cli.Context, cfg *config.Config) error {
+
 	// command line overrides the config file
 	if app.gitUsernameOverride != "" || app.gitPasswordOverride != "" {
+		app.console.Warnf("Warning: the --git-username and --git-password command flags are deprecated and are now configured in the earthly config file under the `git` section\n")
```

Comment:
Nit: I would also remove the backticks around `git` to make it a bit cleaner

Reply:
good point.